grpc-api/src/main/java/io/deephaven/grpc_api/barrage/BarrageMessageProducer.java

```diff
@@ -839,9 +842,17 @@ public void run() {
                         updateSubscriptionsSnapshotAndPropagate();
                     }
                 } catch (final Exception exception) {
-                    // TODO: global error notification core/#55
-                    log.error().append(logPrefix).append("Could not handle barrage update propagation: ")
-                            .append(exception).endl();
+                    synchronized (BarrageMessageProducer.this) {
+                        final StatusRuntimeException apiError = GrpcUtil.securelyWrapError(log, exception);
+                        log.error().append(logPrefix).append("Could not handle barrage update propagation: ")
+                                .append(exception).endl();
+
+                        Streams.concat(activeSubscriptions.stream(), pendingSubscriptions.stream()).distinct()
+                                .forEach(sub -> GrpcUtil.safelyExecute(() -> sub.listener.onError(apiError)));
```

Comment:
probably should use the locked variant so you lock on the sub.listener instance to send the error?